airflow/utils/log/file_task_handler.py

```diff
@@ -56,16 +55,19 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def read(self, task_instance):
+    def read(self, task_instance, try_number):
```

Comment:
From a design point of view this is unclean. `task_instance` is an instantiation of a `task` in time. As such it is bound to a `try_number`, which is now also provides.

I suggest to change the signature of the method to ` def read(self, task_instance, try_number=None)` and the return the latest log if `try_number` isn't specified or maybe even better _all logs_.

Or, use a different signature all together `def read(self, task, execution_date, try_number)`.

On a side note: the try_number thing still feels a bit off. Properly keeping track of _`start_dates`_ (i.e. recording each start_date somewhere) seems more logical. `try_number` in the end is the count of `start_dates`.